public/app/features/dashboard/dashgrid/DashboardLibrary/CommunityDashboardSection.tsx

```diff
@@ -0,0 +1,320 @@
+        orderBy: DEFAULT_SORT_ORDER,
+        direction: DEFAULT_SORT_DIRECTION,
+        page: currentPage,
+        pageSize: COMMUNITY_PAGE_SIZE,
+        includeLogo: INCLUDE_LOGO,
+        includeScreenshots: INCLUDE_SCREENSHOTS,
+        dataSourceSlugIn: ds.type,
+        filter: debouncedSearchQuery.trim() || undefined,
+      });
+
+      // Track analytics on first load
+      if (currentPage === 1 && dashboards.length > 0) {
+        DashboardLibraryInteractions.loaded({
+          numberOfItems: dashboards.length,
+          contentKinds: ['community_dashboard'],
+          datasourceTypes: [ds.type],
+          sourceEntryPoint: 'datasource_page',
+          eventLocation: 'suggested_dashboards_modal_community_tab',
+        });
+      }
+
+      return {
+        dashboards,
+        datasourceType: ds.type,
+      };
+    } catch (err) {
+      console.error('Error loading community dashboards', err);
+      throw err;
+    }
+  }, [datasourceUid, currentPage, debouncedSearchQuery]);
+
+  const styles = useStyles2(getStyles);
+
+  // Determine what to show in results area
+  const dashboards = Array.isArray(response?.dashboards) ? response.dashboards : [];
+  const hasMore = dashboards.length >= COMMUNITY_PAGE_SIZE;
+  const estimatedTotalPages = hasMore ? currentPage + COMMUNITY_PAGINATION_OFFSET : currentPage;
+  const showEmptyState = !loading && (!response?.dashboards || response.dashboards.length === 0);
+  const showError = !loading && error;
+
+  const onUseCommunityDashboard = async (dashboard: GnetDashboard) => {
+    if (response) {
+      DashboardLibraryInteractions.itemClicked({
+        contentKind: 'community_dashboard',
+        datasourceTypes: [response.datasourceType],
+        libraryItemId: String(dashboard.id),
+        libraryItemTitle: dashboard.name,
+        sourceEntryPoint: 'datasource_page',
+        eventLocation: 'suggested_dashboards_modal_community_tab',
+      });
+    }
+
+    try {
+      // Fetch full dashboard from Gcom, this is the JSON with __inputs
+      const fullDashboard = await fetchCommunityDashboard(dashboard.id);
+      const dashboardJson = fullDashboard.json;
+
+      // Parse datasource requirements from __inputs
+      const dsInputs: DataSourceInput[] = dashboardJson.__inputs?.filter(isDataSourceInput) || [];
+
+      // Parse constant inputs - these always need user review
+      const constantInputs = parseConstantInputs(dashboardJson.__inputs || []);
+
+      // Try auto-mapping datasources, considering we could come from "build dashhoard" there should be a datasource
+      // instance selected
+      const mappingResult = tryAutoMapDatasources(dsInputs, datasourceUid || '');
+
+      // Decide whether to show mapping form or navigate directly
+      // Show mapping form if: (a) there are unmapped datasources OR (b) there are constants
+      const needsMapping = mappingResult.unmappedInputs.length > 0 || constantInputs.length > 0;
```

Comment:
we're returning `allMapped` value from `tryAutoMapDatasources` but not using it. Why not to use it here? 
```suggestion
      const needsMapping = !mappingResult.allMapped || constantInputs.length > 0;
```